src/AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -64,6 +69,9 @@ private bool IsGraphQlRequest(HttpContext context)
                 _.UserContext = _options.BuildUserContext?.Invoke(context);
                 _.ExposeExceptions = _options.ExposeExceptions;
                 _.ValidationRules = _options.ValidationRules.Concat(DocumentValidator.CoreRules()).ToList();
+                _documentListners
+                    .ToList()
```

Comment:
Why create another list

Reply:
When you take toList() it instaciates all the registred types and makes a list of it.  
It may be same as just enumerating it. 
do you sugest this approach instead?
```
foreach(var listner in _documentListners){
    _.Listeners.Add(listner)
}

```